napalm/pyIOSXR/iosxr.py

```diff
@@ -348,7 +348,7 @@ def _send_command(
                             delay_factor=delay_factor,
                         )
         else:
-            output += self._netmiko_recv()  # try to read some more
+            output += self._read_channel_timing()  # try to read some more
```

Comment:
@mirceaulinic Any idea when this `else` clause actually is used? I would be inclined to just use Netmiko's `read_channel()` (but I also don't have a good understanding exactly when this else clause would be used).

Also shouldn't this be `self.device._read_channel_timing()`?

Reply:
> Also shouldn't this be self.device._read_channel_timing()?

Ah it should be indeed, great catch! :man_facepalming: 

> Any idea when this else clause actually is used?

I seem to recall this due to the XML agent taking a lot of time to dump the response, and need to wait a little longer, but can't tell for sure, it's been a long time since then. :face_with_head_bandage:

Reply:
Okay, let me look at it just a little bit...I just want to make sure that we shouldn't be using `read_channel()` instead of `read_channel_timing()` (i.e. that read_channel() would be the more reliable way to do this).